### PR TITLE
Support for special characters and remove buggy functions

### PR DESCRIPTION
**Pull request by Alevsk**
- remove the use of encodeURI and encodeURIComponent functions and
  instead use encodeFileName and decodeFileName functions
- support for users with special characters
- support for users with special characters
- support for users with special characters
- fixed incorrectly group list display for policies

Signed-off-by: Lenin Alevski <alevsk.8772@gmail.com>

### CODE REVIEW

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: portal-ui/src/screens/Console/Account/DeleteServiceAccount.tsx [hunk 1]
```diff
@@ -63,7 +64,7 @@ const DeleteServiceAccount = ({
   const onConfirmDelete = () => {
     invokeDeleteApi(
       "DELETE",
-      `/api/v1/service-accounts/${selectedServiceAccount}`
+      `/api/v1/service-accounts/${encodeFileName(selectedServiceAccount)}`
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Account/DeleteServiceAccount.tsx:67]: If we want to use this function for all the REST calls, then I think we must rename it as not everything are file names

File: portal-ui/src/screens/Console/Policies/PolicyDetails.tsx [hunk 2]
```diff
@@ -270,7 +270,7 @@ const PolicyDetails = ({
   ]);
 
   const resetForm = () => {
-    setPolicyName("");
+    // setPolicyName("");
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Policies/PolicyDetails.tsx:273]: Please remove this if useState was removed

File: restapi/admin_groups.go [hunk 3]
```diff
@@ -202,10 +208,16 @@ func getRemoveGroupResponse(session *models.Principal, params groupApi.RemoveGro
 	if err != nil {
 		return ErrorWithContext(ctx, err)
 	}
-	// createad a MinIO Admin Client interface implementation
+	// create a MinIO Admin Client interface implementation
```
Comment (bexsoft) [inline restapi/admin_groups.go:211]:
```suggestion
	// Create a MinIO Admin Client interface implementation
```

**Review by bexsoft — CHANGES_REQUESTED** [review]

File: portal-ui/src/screens/Console/Groups/DeleteGroup.tsx [hunk 4]
```diff
@@ -39,8 +40,8 @@ const DeleteGroup = ({
   const onDelSuccess = () => closeDeleteModalAndRefresh(true);
   const onDelError = (err: ErrorResponseHandler) => {
     setErrorSnackMessage(err);
-    closeDeleteModalAndRefresh(true);
-  }
+    closeDeleteModalAndRefresh(false);
```
[the remark this answers is not in the document]
Comment (Alevsk) [reply portal-ui/src/screens/Console/Groups/DeleteGroup.tsx:43]: This was a bug, we close the modal and redirect the user back to listing groups even if the delete operation failed

File: portal-ui/src/screens/Console/Policies/PolicyDetails.tsx (hunk 2 above)
Comment (Alevsk) [reply portal-ui/src/screens/Console/Policies/PolicyDetails.tsx]: done

**Review by bexsoft — CHANGES_REQUESTED** [review]
Found a couple of issues:

- Add members to group returns 404 in group call
- Set policies to groups has the same issue too
- Add Groups to user
- Saving policies to user returns 404 and don't show the current policies
- Trying to create a service account for a user, redirect you back to buckets list


Screen Shot 2022-05-11 at 23 08 16
Screen Shot 2022-05-11 at 23 07 54
Screen Shot 2022-05-11 at 23 07 16
Screen Shot 2022-05-11 at 23 06 53

**Comment by Alevsk** [comment]
> Found a couple of issues:
> 
> * Add members to group returns 404 in group call
> * Set policies to groups has the same issue too
> * Add Groups to user
> * Saving policies to user returns 404 and don't show the current policies
> * Trying to create a service account for a user, redirect you back to buckets list
> 
> Screen Shot 2022-05-11 at 23 08 16
> 
> Screen Shot 2022-05-11 at 23 07 54
> 
> Screen Shot 2022-05-11 at 23 07 16
> 
> Screen Shot 2022-05-11 at 23 06 53

This issues should be fixed now

**Review by cniackz — APPROVED** [review]
LGTM, it would be nice to increment the coverage threshold accordingly.
But can be done in a separate PR.

**Review by dvaldivia — DISMISSED** [review]

File: portal-ui/src/screens/Console/Users/DeleteUserModal.tsx [hunk 5]
```diff
@@ -51,8 +52,8 @@ const DeleteUserString = ({
   }
 
   const onConfirmDelete = () => {
-    invokeDeleteApi("DELETE", `/api/v1/user?name=${encodeURI(userName)}`, {
-      id: userName,
+    invokeDeleteApi("DELETE", `/api/v1/user/${encodeURLString(userName)}`, {
+      id: encodeURLString(userName),
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Users/DeleteUserModal.tsx:56]: Do we need to send user name twice?

Comment (Alevsk) [reply portal-ui/src/screens/Console/Users/DeleteUserModal.tsx]: this was not need it indeed, its removed now

**Review by bexsoft — DISMISSED** [review]
LGTM

**Review by dvaldivia — DISMISSED** [review]

**Review by bexsoft — APPROVED** [review]

**Review by dvaldivia — APPROVED** [review]

**Pull request merged**